assets/src/components/map.tsx

```diff
@@ -227,88 +229,66 @@ const EventAdder = ({
 }
 
 // #region Auto Center Functionality
-export const autoCenter = (
-  map: LeafletMap,
-  latLngs: LatLngExpression[],
-  pickerContainerIsVisible: boolean
-): void => {
-  if (latLngs.length === 0) {
-    map.setView(defaultCenter, 13)
-  } else if (latLngs.length === 1) {
-    map.setView(latLngs[0], 16)
-  } else if (latLngs.length > 1) {
-    map.fitBounds(Leaflet.latLngBounds(latLngs), {
-      paddingBottomRight: [20, 50],
-      paddingTopLeft: [pickerContainerIsVisible ? 220 : 20, 20],
-    })
-  }
+export type UpdateMapFromPointsFn = (map: LeafletMap, points: LatLng[]) => void
+
+export interface FollowerProps {
+  positions: LatLng[]
+  onUpdate: UpdateMapFromPointsFn
 }
 
-const AutoCenter = ({
-  latLngs,
-  shouldAutoCenter,
-  isAutoCentering,
-}: {
-  shouldAutoCenter: boolean
-  isAutoCentering: MutableRefObject<boolean>
-  latLngs: LatLng[]
-}) => {
+export const Follower = ({
+  positions,
+  isUpdating,
+  onUpdate,
+  shouldFollow = true,
+}: FollowerProps & InteractiveFollowState) => {
   const map = useMap()
-
-  const [{ pickerContainerIsVisible }] = useContext(StateDispatchContext)
-  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(latLngs)
+  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(positions)
 
   if (
-    !equalByElements(latLngs, currentLatLngs, (latLng1, latLng2) =>
-      latLng1.equals(latLng2)
-    )
+    !equalByElements(positions, currentLatLngs, (lhs, rhs) => lhs.equals(rhs))
```

Comment:
question (non-blocking): I realize this mostly from the existing logic so doesn't really need to change, but is there a case when these would not be equal since `currentLatLngs` is initialized to `positions`?

Reply:
I didn't really question this becuase it was the existing logic. I think this is doing a value check rather than using the `Object.is` check that `useEffect` does to keep them updated only when the actual values of `positions` change?

Otherwise, because the `Object.is` function would return `null` for two different instances of arrays of the same data, I think the `useEffect` would fire constantly if it was dependant on `positions`? 
This should probably be checked.

Reply:
@firestack yes, that's all correct as to why it works this way currently. @KaylaBrady if I'm understanding your question correctly, basically on the first render they will be equal, but on subsequent renders the `positions` prop will be updated but `currentLatLngs` will have the old value. Then, the state is only updated if there's an actual change in the contents.